src/aliceVision/sfmData/CameraPose.hpp

```diff
@@ -111,6 +111,23 @@ class CameraPose
         return _removable;
     }
 
+    inline void updateFromExternal(const std::array<double, 6> & data) 
+    {
+        // do not update a camera pose set as Ignored or Constant in the Local strategy
+        if (getState() != EEstimatorParameterState::REFINED)
+        {
+            return;
+        }
```

Comment:
`updateFromExternal` is a very generic name. I was not expecting to see a check related to the optim in it.

Reply:
I will rename the function to updateFromEstimator. I moved out these tool functions from bundle class to landmark and camerapose objects.